src/hooks/useImageLoading/index.tsx

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { useState, useEffect } from 'react'
+
+import { useSharedState } from 'context/state.context'
+
+const _cachedImage = async (img: string, setImgCallback: any) => {
+  const res = await fetch(`https://ipfs.io/ipfs/${img}`)
```

Comment:
Eden has its own IPFS gateway which may be better to use:

https://eden-genesis.mypinata.cloud/ipfs/